repository/asset_allocations_repository.go

```diff
@@ -2,29 +2,75 @@ package repository
 
 import (
 	"context"
+	"database/sql"
+	"time"
 
 	"github.com/jmoiron/sqlx"
 	"github.com/vkhichar/asset-management/contract"
+	"github.com/vkhichar/asset-management/customerrors"
 	"github.com/vkhichar/asset-management/domain"
 )
 
+const (
+	checkIfAssetIsAllocated = "SELECT * FROM asset_allocations WHERE asset_id=$1 ORDER BY id DESC LIMIT 1"
+	createAssetAllocation   = "INSERT INTO asset_allocations(user_id, asset_id, allocated_by, allocated_from) VALUES($1, $2, $3, $4) RETURNING *"
+)
+
 type AssetAllocationsRepository interface {
 	CreateAssetAllocation(ctx context.Context, req contract.CreateAssetAllocationRequest) (*domain.AssetAllocations, error)
 	AssetDeallocation(ctx context.Context, req contract.CreateAssetAllocationRequest) (*domain.AssetAllocations, error)
 }
 
 type assetAllocationsRepo struct {
-	db *sqlx.DB
+	db        *sqlx.DB
+	userRepo  UserRepository
+	assetRepo AssetRepository
 }
 
-func NewAssetAllocationRepository() AssetAllocationsRepository {
+func NewAssetAllocationRepository(uRepo UserRepository, aRepo AssetRepository) AssetAllocationsRepository {
 	return &assetAllocationsRepo{
-		db: GetDB(),
+		db:        GetDB(),
+		userRepo:  uRepo,
+		assetRepo: aRepo,
 	}
 }
 
 func (repo *assetAllocationsRepo) CreateAssetAllocation(ctx context.Context, req contract.CreateAssetAllocationRequest) (*domain.AssetAllocations, error) {
-	return nil, nil
+	user, err := repo.userRepo.GetUserByID(ctx, req.UserId)
```

Comment:
where are you using user details ?

Reply:
Using this query we are just checking if the user exists for whom the request has been made.